diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/libpanel-util/panel-dconf.c mate-panel-1.6.1/mate-panel/libpanel-util/panel-dconf.c
--- mate-panel-1.6.1.OSmanOS/mate-panel/libpanel-util/panel-dconf.c	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/libpanel-util/panel-dconf.c	2013-09-12 11:36:33.863813771 +0300
@@ -111,3 +111,12 @@
 
         return (gchar **) g_array_free (array, FALSE);
 }
+
+void panel_dconf_sync ()
+{
+#ifdef HAVE_DCONF_0_13
+        DConfClient  *client = panel_dconf_client_get ();
+        dconf_client_sync (client);
+        g_object_unref (client);
+#endif
+}
\ Dosya sonunda yenisatır yok.
diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/libpanel-util/panel-dconf.h mate-panel-1.6.1/mate-panel/libpanel-util/panel-dconf.h
--- mate-panel-1.6.1.OSmanOS/mate-panel/libpanel-util/panel-dconf.h	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/libpanel-util/panel-dconf.h	2013-09-12 11:37:41.746811765 +0300
@@ -39,6 +39,8 @@
 gchar **panel_dconf_list_subdirs (const gchar *dir,
                                   gboolean     remove_trailing_slash);
 
+void panel_dconf_sync ();
+
 G_END_DECLS
 
 #endif /* __PANEL_DCONF_H__ */
diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/panel-layout.c mate-panel-1.6.1/mate-panel/panel-layout.c
--- mate-panel-1.6.1.OSmanOS/mate-panel/panel-layout.c	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/panel-layout.c	2013-09-12 11:39:38.313808321 +0300
@@ -122,6 +122,7 @@
     char       *unique_id = NULL;
     char       *path = NULL;
     GSettings  *settings = NULL;
+    GSettings  *panel_settings = NULL;
     char      **keyfile_keys = NULL;
     char       *value_str;
     int         value_int;
@@ -184,6 +185,8 @@
     g_free (path);
 
     keyfile_keys = g_key_file_get_keys (keyfile, group, NULL, NULL);
+    panel_settings = g_settings_new (PANEL_SCHEMA);
+    g_settings_delay (panel_settings);
 
     if (keyfile_keys) {
 
@@ -247,12 +250,9 @@
                                 PANEL_TOPLEVEL_SCREEN_KEY,
                                 set_screen_to);
 
-        GSettings *panel_settings;
-        panel_settings = g_settings_new (PANEL_SCHEMA);
         panel_gsettings_append_strv (panel_settings,
                                      id_list_key,
                                      unique_id);
-        g_object_unref (panel_settings);
 
         retval = TRUE;
     }
@@ -262,6 +262,11 @@
 
     if (settings)
         g_object_unref (settings);
+    
+    if (panel_settings) {
+        g_settings_apply (panel_settings);
+        g_object_unref (panel_settings);
+    }
 
     if (unique_id)
         g_free (unique_id);
diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/panel-profile.c mate-panel-1.6.1/mate-panel/panel-profile.c
--- mate-panel-1.6.1.OSmanOS/mate-panel/panel-profile.c	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/panel-profile.c	2013-09-12 11:40:44.058806378 +0300
@@ -1601,7 +1601,7 @@
 	g_free (key);
 }
 
-static void
+void
 panel_profile_ensure_toplevel_per_screen ()
 {
 	GSList     *toplevels;
diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/panel-profile.h mate-panel-1.6.1/mate-panel/panel-profile.h
--- mate-panel-1.6.1.OSmanOS/mate-panel/panel-profile.h	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/panel-profile.h	2013-09-12 11:41:13.502805508 +0300
@@ -154,6 +154,8 @@
 
 GSettings*  panel_profile_get_attached_object_settings (PanelToplevel *toplevel);
 
+void        panel_profile_ensure_toplevel_per_screen  ();
+
 G_END_DECLS
 
 #endif /* __PANEL_PROFILE_H__ */
diff -Nuar mate-panel-1.6.1.OSmanOS/mate-panel/panel-reset.c mate-panel-1.6.1/mate-panel/panel-reset.c
--- mate-panel-1.6.1.OSmanOS/mate-panel/panel-reset.c	2013-07-12 21:52:29.000000000 +0300
+++ mate-panel-1.6.1/mate-panel/panel-reset.c	2013-09-12 11:42:43.250802857 +0300
@@ -28,19 +28,24 @@
 
 #include <stdlib.h>
 #include <glib.h>
+#include <gio/gio.h>
 #include "panel-reset.h"
 #include "panel-schemas.h"
+#include "panel-profile.h"
 #include <libpanel-util/panel-dconf.h>
 
 void
 panel_reset()
-{
+{       
+        /* reset configuration */
 	panel_dconf_recursive_reset (PANEL_GENERAL_PATH, NULL);
 	panel_dconf_recursive_reset (PANEL_TOPLEVEL_PATH, NULL);
 	panel_dconf_recursive_reset (PANEL_OBJECT_PATH, NULL);
-
-	/* TODO: send a dbus message to mate-panel, if active, to reload the panel
-	 * configuration */
+        panel_dconf_sync ();
+        
+        /* create default panels */
+       panel_profile_ensure_toplevel_per_screen ();
+       g_settings_sync ();
 }
 
 #endif /* !__PANEL_RESET_C__ */
